android/src/main/kotlin/com/mapbox/maps/mapbox_maps/Extentions.kt

```diff
@@ -263,6 +270,74 @@ fun RenderedQueryOptions.toRenderedQueryOptions(): com.mapbox.maps.RenderedQuery
   return com.mapbox.maps.RenderedQueryOptions(layerIds, filter?.toValue())
 }
 
+fun FeaturesetFeatureId.toFeaturesetFeatureId(): com.mapbox.maps.FeaturesetFeatureId {
+  return com.mapbox.maps.FeaturesetFeatureId(id, namespace)
+}
+
+fun FeaturesetDescriptor.toFeatureSetDescriptor(): com.mapbox.maps.FeaturesetDescriptor {
+  return com.mapbox.maps.FeaturesetDescriptor(featuresetId, importId, layerId)
+}
+
+@OptIn(MapboxExperimental::class)
+fun FeaturesetDescriptor.toTypedFeaturesetDescriptor(): TypedFeaturesetDescriptor<*, *> {
+  featuresetId?.let {
+    return TypedFeaturesetDescriptor.Featureset(
+      featuresetId, importId
+    )
+  }
+  return TypedFeaturesetDescriptor.Layer(
+    layerId!! // If there is not a featuresetId there will be a layerId
+  )
+}
+
+fun FeaturesetQueryTarget.toFeaturesetQueryTarget(): com.mapbox.maps.FeaturesetQueryTarget {
+  return com.mapbox.maps.FeaturesetQueryTarget(featureset.toFeatureSetDescriptor(), filter?.let { Expression.fromRaw(filter) }, id)
+}
+
+@OptIn(MapboxExperimental::class)
+fun Map<String, Any?>.toFeatureState(): com.mapbox.maps.interactions.FeatureState {
+  val map = this
+  return FeatureState {
+    for ((key, value) in map) {
+      when (value) {
+        is String -> {
+          addStringState(key, value)
+        }
+        is Long -> {
+          addLongState(key, value)
+        }
+        is Double -> {
+          addDoubleState(key, value)
+        }
+        is Boolean -> {
+          addBooleanState(key, value)
+        }
+        else -> throw (RuntimeException("Unsupported (key, value): ($key, $value)"))
+      }
+    }
+  }
+}
+
+@OptIn(MapboxExperimental::class)
+@SuppressLint("RestrictedApi")
+fun FeaturesetFeature.toFeaturesetFeature(): com.mapbox.maps.interactions.FeaturesetFeature<FeatureState> {
```

Comment:
@pjleonard37 if I understand it right, this function is not needed. From what I see you use it internally only to map Flutter `FeaturesetFeature` to the Android one. However in Android (and most likely in iOS) we provide low-level API for `get / set / remove FeatureState` that take `TypedFeaturesetDescriptor` and `FeaturesetFeatureId` as parameters instead of `FeaturesetFeature`.

If you'll use those - you can remove `toFeaturesetFeature()` from what I see and also get rid of those nasty JSON conversions (which are actually not needed at all).

Reply:
example: third overload [here](https://docs.mapbox.com/android/maps/api/11.8.0-rc.1/mapbox-maps-android/com.mapbox.maps/-mapbox-map/set-feature-state.html?query=fun%20%3CFS%20:%20FeatureState%3E%20setFeatureState(descriptor:%20TypedFeaturesetDescriptor%3CFS,%20*%3E,%20id:%20FeaturesetFeatureId,%20state:%20FS,%20callback:%20FeatureStateOperationCallback%20=%20FeatureStateOperationCallback%20{%20}):%20Cancelable) is the one that I would like to be used in Flutter

Reply:
Interesting, I'll need to think about this a bit 🤔 

>  get rid of those nasty JSON conversions
This would be great 👍